core/app/models/concerns/spree/user_address.rb

```diff
@@ -12,15 +12,15 @@ module UserAddress
       accepts_nested_attributes_for :ship_address, :bill_address
 
       def persist_order_address(order)
-        b_address = self.bill_address || self.build_bill_address
-        b_address.attributes = order.bill_address.attributes.except('id', 'updated_at', 'created_at')
-        b_address.save
```

Comment:
Oh, and: Try this code on the console. Doing `b_address.attributes = order.bill_address.attributes.except('id', 'updated_at', 'created_at')` is a no-op. That doesn't do anything. I believe this code is intended to clone the address but doesn't?